pkg/distributor/distributor.go

```diff
@@ -339,6 +398,59 @@ func (d *Distributor) Push(ctx context.Context, req *logproto.PushRequest) (*log
 	}
 }
 
+func min(x1, x2 int) int {
```

Comment:
Out of interest, do we have this because `math.Min` only accepts `float64`?

Reply:
yeah. I believe in the near future we might migrate to a generic implementation though (because of... generics)